nifi-commons/nifi-utils/src/main/java/org/apache/nifi/stream/io/BufferedInputStream.java

```diff
@@ -16,19 +16,445 @@
  */
 package org.apache.nifi.stream.io;
 
+import java.io.IOException;
 import java.io.InputStream;
 
 /**
  * This class is a slight modification of the BufferedInputStream in the java.io package. The modification is that this implementation does not provide synchronization on method calls, which means
  * that this class is not suitable for use by multiple threads. However, the absence of these synchronized blocks results in potentially much better performance.
```

Comment:
Well, FWIW, I just read the 5GB file 1B at the time and here are the numbers (in millis) of elapsed time:
```
121714
124480
123031
121378
```
The first two using java.io, the second using NIFI's.
There are differences but as you can see non can be attributed to sync especially if you look at the first and last number. In fact one can argue that second, third and fourth run was already on a pre-warmed page cache. 
All I am questioning if it's worth for us to maintain extra code when one is available in core Java?